### PR TITLE
Catch rdrs go runtime error

### PR DESCRIPTION
**Pull request by kennethmhc**
In general, when there are runtime errors, we should catch and log it and response with an error message.

### CODE REVIEW

File: storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go [hunk 1]
```diff
@@ -17,16 +17,29 @@
...
 }
 
-func Handle(h Handler, apiKey *string, request interface{}, response interface{}) (int, error) {
-
+func Handle(h Handler, apiKey *string, request interface{}, response interface{}) (statusCode int, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			statusCode = http.StatusInternalServerError
+			err = errors.New(`{"message": "Internal server error."}`)
+			log.Error(r.(error).Error())
```
Comment (kennethmhc) [inline storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go]: is this thread safe?

Comment (olapiv) [reply storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go]: You mean the log library? Should be, yes.

**Review by olapiv — DISMISSED** [review]

**Comment by olapiv** [comment]
Good idea, I think it’s important to have some kind of recovery at some stage in the code. Have you tested that the return values work if the Go code panics?

File: storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go [hunk 2]
```diff
@@ -17,16 +17,29 @@
...
 	Execute(request interface{}, response interface{}) (int, error)
 }
 
-func Handle(h Handler, apiKey *string, request interface{}, response interface{}) (int, error) {
-
+func Handle(h Handler, apiKey *string, request interface{}, response interface{}) (statusCode int, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			statusCode = http.StatusInternalServerError
+			err = errors.New(`{"message": "Internal server error."}`)
```
Comment (olapiv) [inline storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go:38]: This is our current error return format?

File: storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go (hunk 1 above)
Comment (olapiv) [reply storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go]: You can ignore the comment in the GO code regarding the thread-safe logger. What I meant there was just to use a Go logger instance/object, like zap.logger.

File: storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go (hunk 2 above)
Comment (kennethmhc) [reply storage/ndb/rest-server/rest-api-server/internal/handlers/handler.go:38]: This is up for discussion.

**Comment by kennethmhc** [comment]
@olapiv yes, return value works when there is a  panic in Go.

**Comment by kennethmhc** [comment]
@olapiv @smkniazi please take a look

**Pull request closed without merging**